django/core/checks/security/base.py

```diff
@@ -208,3 +213,8 @@ def check_xframe_deny(app_configs, **kwargs):
 @register(Tags.security, deploy=True)
 def check_allowed_hosts(app_configs, **kwargs):
     return [] if settings.ALLOWED_HOSTS else [W020]
+
+
+@register(Tags.security, deploy=True)
+def check_type_of_allowed_hosts(app_configs, **kwargs):
+    return [] if not isinstance(settings.ALLOWED_HOSTS, str) else [W022]
```

Comment:
I think we should check it's iterable AND not a string, there's always the possibility of other mistakes than the one that lead to the ticket, e.g. missing brackets on a function call